src/ua.c

```diff
@@ -752,8 +755,15 @@ static int uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 
 	err |= mbuf_write_str(buf, uri);
 
-	/* Append domain if missing */
-	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL)) {
+	/* Append domain if missing and uri is not IP address */
+	/* check if uri is valid IPv4 address */
+	uri_is_ip = inet_pton(AF_INET, uri, &(sa.sin_addr));
+#if HAVE_INET6
+	/* check if uri is valid IPv6 address */
+	uri_is_ip |= inet_pton(AF_INET6, uri, &(sa.sin_addr));
+#endif
+	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL) &&
+		1 != uri_is_ip) {
```

Comment:
it is better to use the `sa` functions from libre, they are portable and handle
both IPv4 and IPv6.

for example:

```
	struct sa tmp;

	uri_is_ip = (0 == sa_set_str(&tmp, uri, 0));
```